File: Python/program/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):

    __jenisMotor = ""
    __kapasitasTangki = ""

    def __init__(self, platNomor="", merk="", tahunProduksi=-1, warna="", jenisMotor="", kapasitasTangki=""):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        self.__jenisMotor = jenisMotor
        self.__kapasitasTangki = kapasitasTangki
    
    def getJenisMotor(self):
        return self.__jenisMotor

    def setJenisMotor(self, jenisMotor):
        self.__jenisMotor = jenisMotor

    def getKapasitasTangki(self):
        return self.__kapasitasTangki

    def setKapasitasTangki(self, kapasitasTangki):
        self.__kapasitasTangki = kapasitasTangki
    
    def getByColName(self, colName=""):
        if colName == "platNomor":
            return self.getPlatNomor()
        elif colName == "merk":
            return self.getMerk()
        elif colName == "tahunProduksi":
            return self.getTahunProduksi()
        elif colName == "warna":
            return self.getWarna()
        elif colName == "jenisMotor":
            return self.getJenisMotor()
        elif colName == "kapasitasTangki":
            return self.getKapasitasTangki()

File: CPP/Motorcycle.cpp
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Motorcycle : public Vehicle{
    private:
        int kapasitasTangki;
        string jenisMotor;
    public:
        Motorcycle(){

        }
        Motorcycle(string platNomor, string merk, int tahunProduksi, string warna, int kapasitasTangki) : Vehicle(platNomor, merk, tahunProduksi, warna){
            this->kapasitasTangki = kapasitasTangki;
            this->jenisMotor = "";
        }
        Motorcycle(string platNomor, string merk, int tahunProduksi, string warna, int kapasitasTangki, string jenisMotor) : Vehicle(platNomor, merk, tahunProduksi, warna){
            this->kapasitasTangki = kapasitasTangki;
            this->jenisMotor = jenisMotor;
        }

        void setKapasitasTangki(int kapasitasTangki){
            this->kapasitasTangki = kapasitasTangki;
        }
        void setJenisMotor(string jenisMotor){
            this->jenisMotor = jenisMotor;
        }

        int getKapasitasTangki(){
            return this->kapasitasTangki;
        }
        string getJenisMotor(){
            return this->jenisMotor;
        }

        string getByColName(string colName){
            if(colName == "platNomor"){
                return this->getPlatNomor();
            }
            else if(colName == "merk"){
                return this->getMerk();
            }
            else if(colName == "warna"){
                return this->getWarna();
            }
            else if(colName == "tahunProduksi"){
                return to_string(this->getTahunProduksi());
            }
            else if(colName == "jenisMotor"){
                return this->getJenisMotor();
            }
            else if(colName == "kapasitasTangki"){
                return to_string(this->getKapasitasTangki());
            }
        }

        ~Motorcycle(){

        }
};